src/plugins/intel_gpu/src/plugin/transformations/disable_f16_comp_for_periodic_funcs.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <vector>
+
+#include "openvino/op/util/sub_graph_base.hpp"
+#include "transformations/rt_info/fused_names_attribute.hpp"
+#include "transformations/rt_info/primitives_priority_attribute.hpp"
+#include "transformations/utils/utils.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/op/sin.hpp"
+#include "openvino/op/sinh.hpp"
+#include "openvino/op/cos.hpp"
+#include "openvino/op/cosh.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/util/scatter_base.hpp"
+#include "openvino/op/util/scatter_nd_base.hpp"
+#include "openvino/op/util/scatter_elements_update_base.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/split.hpp"
+#include "openvino/op/strided_slice.hpp"
+#include "openvino/op/tile.hpp"
+#include "openvino/op/identity.hpp"
+#include "openvino/op/pad.hpp"
+#include "openvino/op/util/gather_base.hpp"
+#include "openvino/op/util/gather_nd_base.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "disable_f16_comp_for_periodic_funcs.hpp"
+
+#include <unordered_map>
+#include <typeindex>
+
+static bool is_non_value_modifying_node(const std::shared_ptr<ov::Node>& node) {
+    static const std::unordered_map<std::string, bool> node_behavior_registry = {
+        {ov::op::v1::Reshape::get_type_info_static().name, true},
+        {ov::op::v1::Transpose::get_type_info_static().name, true},
+        {ov::op::util::ScatterBase::get_type_info_static().name, true},
+        {ov::op::util::ScatterNDBase::get_type_info_static().name, true},
+        {ov::op::util::ScatterElementsUpdateBase::get_type_info_static().name, true},
+        {ov::op::v8::Slice::get_type_info_static().name, true},
+        {ov::op::v1::Broadcast::get_type_info_static().name, true},
+        {ov::op::v0::Concat::get_type_info_static().name, true},
+        {ov::op::v1::Split::get_type_info_static().name, true},
+        {ov::op::v1::StridedSlice::get_type_info_static().name, true},
+        {ov::op::v0::Tile::get_type_info_static().name, true},
+        {ov::op::v16::Identity::get_type_info_static().name, true},
+        {ov::op::v1::Pad::get_type_info_static().name, true},
+        {ov::op::util::GatherNDBase::get_type_info_static().name, true},
+        {ov::op::util::GatherBase::get_type_info_static().name, true},
+        {ov::op::v15::Squeeze::get_type_info_static().name, true},
+        {ov::op::v0::Unsqueeze::get_type_info_static().name, true},
```

Comment:
why is `true` needed here?
also, the name `node_behavior_registry` may be too general name here, if `true` is removed.